.github/workflows/ci.yml

```
@@ -0,0 +1,48 @@
+# This workflow will build a Java project with Maven
+# For more information see: https://help.github.com/actions/language-and-framework-guides/building-and-testing-java-with-maven
+
+name: JBoss Modules CI
+
+on:
+  pull_request:
+    types: [opened, synchronize, reopened, ready_for_review]
+
+jobs:
+  build-test-matrix:
+    name: JDK${{ matrix.jdk }}-${{ matrix.impl}}-${{ matrix.os }}
+    runs-on: ${{ matrix.os }}
+    strategy:
+      fail-fast: false
+      matrix:
+        os: [ubuntu-latest, windows-latest]
+        jdk: [11]
+        impl: [hotspot, openj9]
+    steps:
+    - uses: actions/checkout@v2
+    # test runs use -Djava8.home=... as well, so we need both JDKs
+    - name: Set up JDK 8
+      uses: AdoptOpenJDK/install-jdk@v1
+      with:
+        version: 8
+        impl: ${{ matrix.impl }}
+    - name: Store JDK8 env (Linux)
+      if: contains(matrix.os, 'ubuntu')
+      run: |
+        echo "::set-env name=JAVA8_HOME::$(echo $JAVA_HOME)"
+    - name: Store JDK8 env (Windows)
+      if: contains(matrix.os, 'windows')
+      run: |
+        echo "::set-env name=JAVA8_HOME::$env:JAVA_HOME"
+    - name: Set up JDK ${{ matrix.jdk }}
+      uses: AdoptOpenJDK/install-jdk@v1
+      with:
+        version: ${{ matrix.jdk }}
+        impl: ${{ matrix.impl }}
+    - name: Run Build & Test
+    # -D needs to be escaped in powershell, use --define= instead
+      run: mvn -U -B -fae clean install --define=java8.home=${{ env.JAVA8_HOME }}
```

```suggestion
      run: mvn -U -B -fae clean install --define=java8.home=${{ env.JAVA_HOME_8 }}
```

Assuming it works on Windows that is.  If so, you could delete the "Store JDK8 env" bits...

---

It would be good to also add java9 and java10 TBH.

---

+1 on the var rename, and yeah will add 9 & 10 as well